cosmos/operators/local.py

```diff
@@ -336,17 +337,25 @@ def _upload_sql_files(self, tmp_project_dir: str, resource_type: str) -> None:
         elapsed_time = time.time() - start_time
         self.log.info("SQL files upload completed in %.2f seconds.", elapsed_time)
 
-    def _delete_sql_files(self, tmp_project_dir: Path, resource_type: str) -> None:
+    def _delete_sql_files(self) -> None:
+        """Deletes the entire run-specific directory from the remote target."""
         dest_target_dir, dest_conn_id = self._configure_remote_target_path()
-        source_run_dir = Path(tmp_project_dir) / f"target/{resource_type}"
-        files = [str(file) for file in source_run_dir.rglob("*") if file.is_file()]
+        if not dest_target_dir or not dest_conn_id:
```

Comment:
@tuantran0910 It seems we didn't write a test that covers this use case. Could you please add it?